axelrod/fingerprint.py

```diff
@@ -0,0 +1,206 @@
+from axelrod.interaction_utils import compute_final_score_per_turn as cfspt
+from collections import OrderedDict
+
+
+class Fingerprint():
+    def __init__(self):
+        pass
+
+    def plot(self):
+        pass
+
+
+class AshlockFingerprint(Fingerprint):
+    def __init__(self, strategy, probe):
+        """
+        Parameters
+        ----------
+        strategy : class
+            A class that must be descended from axelrod.strategies.
+        probe : class
+            A class that must be descended from axelrod.strategies
+        """
+        self.strategy = strategy
+        self.probe = probe
+
+    def create_probe_coords(self, granularity):
+        """Creates a set of coordinates over the unit square.
+
+        Construncts (x, y) coordinates that are separated by a step equal to
+        `granularity`. The coordinates are over the unit squeare which implies
+        that the number of points created will be 1/`granularity`^2.
+
+        Parameters
+        ----------
+        granularity : float
+            The seperation between each coordinate. Smaller granularity will
+            produce more coordinates that will be closer together.
+
+        Returns
+        ----------
+        probe_coords : ordered dictionary
+            An Ordered Dictionary where the keys are tuples representing each
+            coordinate, eg. (x, y). The value is automatically set to `None`.
+        """
+        coordinates = list(product(np.arange(0, 1, granularity),
+                                   np.arange(0, 1, granularity)))
+
+        probe_coords = OrderedDict.fromkeys(coordinates)
+        return probe_coords
+
+    def create_probes(self, probe, granularity):
+        """Creates a set of probe strategies over the unit square.
+
+        Construncts probe strategies that correspond to (x, y) coordinates. The
+        precision of the coordinates is determined by `granularity`. The probes
+        are created using the `JossAnnTransformer`.
+
+        Parameters
+        ----------
+        granularity : float
+            The seperation between each coordinate. Smaller granularity will
+            produce more coordinates that will be closer together.
+        probe : class
+            A class that must be descended from axelrod.strategies.
+
+        Returns
+        ----------
+        probe_dict : ordered dictionary
+            An Ordered Dictionary where the keys are tuples representing each
+            coordinate, eg. (x, y). The value is a `JossAnnTransformer` with
+            parameters that correspond to (x, y).
+        """
+        probe_dict = self.create_probe_coords(granularity)
+        for coord in probe_dict.keys():
+            x, y = coord
+            if x + y > 1:
+                probe_dict[coord] = JossAnnTransformer((1 - y, 1 - x))(probe)()
+            else:
+                probe_dict[coord] = JossAnnTransformer((x, y))(probe)()
```

Comment:
For me, the clue in this one is that it doesn't use `self` anywhere within the method.